README.md

```diff
@@ -137,10 +137,18 @@ Configures Lambda event sources (triggers) specific to each environment.
 
 ### Process a Lambda Event
 
+This lambda has three events:
+- DiscoveryEvent.json, simulating a hold request from Discovery UI
+- RecapEvent.json, simulating a hold request from Recap UI
+- EddEvent.json, simulating an edd request from Discovery UI.
+
+HoldRequestConsumer uses the absence/presence of the deliveryLocation property to classify the event as coming from Discovery UI/Recap UI. Only an event from Discovery UI will be processed to the Recap API. The RecapEvent will be filtered out by the HoldRequestConsumer. In case of an event from Discovery UI, if there are errors or in case of an EDD request, the HoldRequestConsumer will write an event to the HoldRequestResult Kinesis Stream. So, of the three events above, only EddEvent.json should result in an event hitting the HoldRequestResultConsumer and an email.
+
```

Comment:
As discussed and collaborated on, here's more detail on these events and the three different flows generally - for possible inclusion:

### DiscoveryEvent: simulating a hold request from Discovery UI

1. Because the event lacks a deliveryLocation, HoldRequestConsumer interprets the event as originating from the Discovery UI
2. HoldRequestConsumer calls the Recap API
3. If the Recap API returns a success response, HoldRequestConsumer does nothing.
4. If the Recap API returns a failure response, HoldRequestConsumer records the error in HoldRequestResult stream.

### RecapEvent: simulating a hold request from Recap UI

1. Because the event contains a deliveryLocation, HoldRequestConsumer interprets the event as a Recap UI originating event and filters it out (i.e. does nothing with it).   (Presumably ReCAP already knows about the hold request represented.)

### EddEvent: simulating an edd request from Discovery UI

1. An EDD event causes the Consumer to hit the ReCAP API and post a version of the result to the HoldRequestResult stream
2. The HoldRequestResultConsumer reads the HoldRequestResult stream looking specifically for EDD requests in-process, and sends an email via SES.
3. If SES returns success, the HoldRequest object has `.processed` set to TRUE and a PATCH is sent to HoldRequestService. (HoldRequestService updates the record and broadcasts the change via the HoldRequest stream - which is ignored by HoldRequestConsumer because processed === TRUE.)

Reply:
@gkallenberg can you take a look at this and let us know if you recall any additional we should say/know about how data flows downstream from (and back into) this service? Thanks

Reply:
Yes, for the DiscoveryEvent on ReCAP API success and for the RecapEvent, , the request is posted to the RecapHoldRequestService which publishes the request to its stream which is read by the RecapHoldRequestConsumer.

The RecapHoldRequestConsumer places a hold in Sierra via the Sierra API for NYPL owned items or creates a temporary record via NCIP to track non-NYPL items.

On success, the RecapHoldRequestConsumer publishes a message to the HoldRequestResultStream which sends a PATCH to the HoldRequestServices and attempts to send an email via SES.

On failure, the RecapHoldRequestConsumer logs the failure and publishes to the HoldRequestResultStream which sends a PATCH to the HoldRequestService with `status === false` and `processed === true`.

As stated under the EDD event, updates to the HoldRequestService which are published to its stream are ignored by the HoldRequestConsumer.

I believe that's the rest of the story. I'm not quite sure about the details for the RecapHoldRequestConsumer. You may want to check with Kris or Kevin.

For #4 under the DiscoveryEvent, the PATCH sent by the HoldRequestResultConsumer includes `processed === false`